games/static/games/proto.js

```diff
@@ -204,18 +199,32 @@ class Board {
         }
     }
 
+    determineDirection() {
+        const aliensAtLeftBorder = this.squares[this.alienLocations[0]].isLeftBorder()
+        const aliensAtRightBorder = this.squares[this.alienLocations[this.alienLocations.length - 1]].isRightBorder()
+        const Left = -1
+        const Right = 1
+        if ((aliensAtLeftBorder && this.direction === Left) || (aliensAtRightBorder && this.direction === Right)) {
+            this.direction = this.width
```

Comment:
a short note about it.
basicly its ok
but
you see in a lot of clean code practices that we tend to seperate the calculation from mutation
calculte the new direction, get the result, and only then override the class member (aka mutate it).
this is a better practice, and easier to test